conda_build_prepare/prepare.py

```diff
@@ -136,10 +137,30 @@ def _try_to_get_git_output(cmd_string):
         YAML().dump(metadata, meta)
 
 
+def _set_date_env_vars(host_git_repo):
+    if 'DATE_NUM' not in os.environ or 'DATE_STR' not in os.environ:
+        try:
+            date_num = git_get_heads_time(host_git_repo, '%Y%m%d%H%M%S')
+            print(f"Setting '{date_num}' as DATE_NUM")
+            os.environ['DATE_NUM'] = date_num
+
+            date_str = git_get_heads_time(host_git_repo, '%Y%m%d_%H%M%S')
+            print(f"Setting '{date_str}' as DATE_STR.")
+            os.environ['DATE_STR'] = date_str
+        except subprocess.CalledProcessError:
+            print()
+            print('WARNING: Failed to set default DATE_NUM and DATE_STR.')
+            print('  This is normal if the recipe isn\'t in a git repository.')
```

Comment:
can we check if we're a git repo? Because this does not really tell if we fail or not here

Reply:
Apart from using additional modules for git I think I can only call `git status` and see whether it succeeds. I was thinking about using such a module, e.g. GitPython, but then it might be reasonable to use it for all git interactions which is a big change.

What do you think, maybe I should simply start using it now from the new code introduced with this PR?

Reply:
naah, not that important. I would consider adding a `git status` helper, that just calls it and checks if it fails or not.

Or maybe `git rev-parse --is-inside-work-tree`